src/components/user/services/user.service.v2.integration.test.js

```diff
@@ -0,0 +1,163 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { ObjectId } = require('mongoose').Types;
+const appPromise = require('../../../app');
+const testObjects = require('../../../../test/services/helpers/testObjects')(appPromise);
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('user service v2', function test() {
+	let app;
+	let server;
+	let accountModelService;
+	let usersModelService;
+
+	before(async () => {
+		app = await appPromise;
+		server = await app.listen(0);
+		accountModelService = app.service('accountModel');
+		usersModelService = app.service('usersModel');
+	});
+
+	after(async () => {
+		await server.close();
+	});
+
+	const getAdminToken = async (schoolId = undefined) => {
+		const adminUser = await testObjects.createTestUser({ roles: ['administrator'], schoolId });
+		const credentials = { username: adminUser.email, password: `${Date.now()}` };
+		await testObjects.createTestAccount(credentials, 'local', adminUser);
+		const token = await testObjects.generateJWT(credentials);
+		return token;
+	};
+
+	const getPermissionToken = async (schoolId, permissions = []) => {
+		const currentUserRole = `currentUserPermission`;
+		await testObjects.createTestRole({
+			name: currentUserRole,
+			permissions,
+		});
+		const currentUser = await testObjects.createTestUser({
+			firstName: 'deleteUser',
+			roles: [currentUserRole],
+			schoolId,
+		});
+		const credentials = { username: currentUser.email, password: `${Date.now()}` };
+		await testObjects.createTestAccount(credentials, 'local', currentUser);
+		const token = await testObjects.generateJWT(credentials);
+		return token;
+	};
+
+	describe('API tests', () => {
+		it('When an authorized user deletes a student and returns success', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const user = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const token = await getAdminToken(schoolId);
+			const request = chai
+				.request(app)
+				.delete(`/users/v2/admin/student/${user._id.toString()}`)
+				.query({ userId: user._id.toString() })
+				.set('Accept', 'application/json')
+				.set('Authorization', token)
+				.set('Content-type', 'application/json');
+			const response = await request.send();
+			expect(response.status).to.equal(200);
+			expect(response.body.userId).to.deep.equal(user._id.toString());
+		});
+
+		it('Fails when not authorized user deletes a student', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			// const params = await testObjects.generateRequestParamsFromUser(admin);
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+			const user = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const token = await testObjects.generateJWTFromUser(teacher);
```

Comment:
teachers permission may change in the future. please either explicitly name the teacher in the title of the test, or create a new role without the permission.